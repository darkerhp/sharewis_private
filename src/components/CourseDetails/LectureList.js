import React from 'react';
import ReactNative from 'react-native';

import BaseStyles from '../../baseStyles';

import LectureItem from './LectureItem';
import Section from './Section';

const { PropTypes } = React;
const { View, StyleSheet } = ReactNative;

const styles = StyleSheet.create({
  container: {
    borderColor: BaseStyles.borderColor,
    borderTopWidth: 1,
  },
});

const renderLecture = (key, courseId, lecture, handlePressLecture, handlePressDownload,
                       fetchDownloadStatus) => (
  lecture.kind === 'section'
    ? <Section key={key} lecture={lecture} />
    : <LectureItem
      key={key}
      courseId={courseId}
      currentLecture={lecture}
      handlePressLecture={handlePressLecture}
      handlePressDownload={handlePressDownload}
      fetchDownloadStatus={fetchDownloadStatus}
    />
);

const LectureList = ({
  courseId,
  lectures,
  containerStyleId,
  handlePressLecture,
  handlePressDownload,
  fetchDownloadStatus }) =>
  <View style={[styles.container, containerStyleId]}>
    {lectures.map((lecture, i) => renderLecture(
      i,
      courseId,
      lecture,
      handlePressLecture,
      handlePressDownload,
      fetchDownloadStatus)
    )}
  </View>;

LectureList.propTypes = {
  lectures: PropTypes.arrayOf(PropTypes.shape({
    /* eslint-disable react/no-unused-prop-types */
    order: PropTypes.number,
    title: PropTypes.string,
    kind: PropTypes.string,
    duration: PropTypes.number,
    isCompleted: PropTypes.bool,
    type: PropTypes.string,
    /* eslint-enable react/no-unused-prop-types */
  })).isRequired,
  handlePressLecture: PropTypes.func.isRequired,
  containerStyleId: PropTypes.number.isRequired,
  handlePressDownload: PropTypes.func.isRequired,
};

export default LectureList;
