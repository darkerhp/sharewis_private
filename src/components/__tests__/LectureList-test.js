/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import LectureList from '../CourseDetails/LectureList';

jest.mock('../CourseDetails/LectureItem', () => 'LectureItem');

describe('LectureList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <LectureList
        courseId={0}
        lectures={[]}
        containerStyleId={0}
        handlePressDelete={() => null}
        handlePressDownload={() => null}
        handlePressLecture={() => null}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
