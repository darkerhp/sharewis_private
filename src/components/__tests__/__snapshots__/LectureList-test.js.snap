exports[`LectureList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#ddd",
        "borderTopWidth": 1
      },
      0
    ]
  } />
`;
